app/uk/gov/hmrc/emailverification/models/PasscodeDoc.scala

```diff
@@ -17,12 +17,18 @@
 package uk.gov.hmrc.emailverification.models
 
 import org.joda.time.DateTime
-import play.api.libs.json.{Format, Json, OFormat}
+import play.api.libs.json.{Format, JsObject, JsResult, JsValue, Json, OFormat}
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
-case class PasscodeDoc(sessionId:String, email: String, passcode: String, expireAt: DateTime)
+case class PasscodeDoc(sessionId:String, email: String, passcode: String, expireAt: DateTime, passcodeAttempts:Int=0, emailAttempts:Int=0)
 
 object PasscodeDoc {
   implicit val dateTimeFormats: Format[DateTime] = ReactiveMongoFormats.dateTimeFormats
-  implicit val format: OFormat[PasscodeDoc] = Json.format[PasscodeDoc]
+  implicit val passcodeReads = Json.reads[PasscodeDoc]
+  implicit val passcodeWrites = Json.writes[PasscodeDoc]
+  implicit val format = new OFormat[PasscodeDoc]() {
+    override def reads(json: JsValue): JsResult[PasscodeDoc] = passcodeReads.reads(json)
+    override def writes(passcodeDoc: PasscodeDoc): JsObject = passcodeWrites.writes(passcodeDoc)
+  }
```

Comment:
would `implicit val format = Json.format[PasscodeDoc]` do the same thing? (although if it requires that dateTimeFormat above to work correctly, I'd prefer it was more explicit than using the macros)

Reply:
👍